pkg/performanceprofile/controller/performanceprofile_controller.go

```diff
@@ -422,6 +426,29 @@ func (r *PerformanceProfileReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return reconcile.Result{}, nil
 	}
 
+	// Update the config.openshift.io/node object with the desired cgroupsv1 mode.
+	// (TODO) This code can be removed in the future when the cgroupsv2 is supported
+	key := types.NamespacedName{
+		Name: nodeCfgName,
+	}
+	nodeCfg := &apiconfigv1.Node{}
+	nodeCfg.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "config.openshift.io",
+		Version: "v1",
+		Kind:    "Node",
+	})
+	err = r.Client.Get(context.Background(), key, nodeCfg)
```

Comment:
This is also executed during every reconciliation, right? Will the API team complain the number of requests from NTO got higher?

Reply:
@harche , @haircommander Do you have any thoughts on this?

Reply:
I think this needs to be addressed, unnecessary overhead. This really should only need to be done at startup

Reply:
We are looking for pointers on how to do that in NTO, https://github.com/openshift/cluster-node-tuning-operator/pull/737#issuecomment-1655797423

Reply:
Although this change will only get the nodes config object and only if it set to v2, it will try to set it to v1. This will not have any effect if the cluster is already in v1.

Reply:
The number of request to API (to fetch the nodes config object) would be proportional to the number of times performance profile is changed.

Reply:
We can watch this object using predicates and act accordingly in case it's set to v2
https://github.com/openshift/cluster-node-tuning-operator/blob/master/pkg/performanceprofile/controller/performanceprofile_controller.go#L149 
BTW this code is only temporarily right?  until NTO catches up with cgroupsv2

Reply:
I don't think we need to add an additional watch here.  
What we see here are additional API calls (get/update) added into the reconcile loop (which has more triggers than only performance profile changes) 
Couldn't this be applied in the upper level ? 

for example, where we still have the upgrade code for removing PAO old operator:
https://github.com/openshift/cluster-node-tuning-operator/blob/96b067113eca80a24b649fe044ec0047b580eccb/cmd/cluster-node-tuning-operator/main.go#L138

Reply:
> This is also executed during every reconciliation, right? Will the API team complain the number of requests from NTO got higher?

@MarSik  Adding a Get operation in the reconcile loop is that expensive ?  assuming we cannot add this at startup and MUST have a performance profile present inorder to switch to v1